src/main/java/ch/njol/skript/lang/Section.java

```diff
@@ -66,6 +91,47 @@ protected void loadCode(SectionNode sectionNode) {
 		}
 	}
 
+	/**
+	 * Loads the code in the given {@link SectionNode},
+	 * appropriately modifying {@link ParserInstance#getCurrentSections()}.
+	 *
+	 * This method differs from {@link #loadCode(SectionNode)} in that it
+	 * is meant for code that will be executed in a different event.
+	 *
+	 * @param sectionNode The section node to load.
+	 * @param name The name of the event(s) being used.
+	 * @param events The event(s) during the section's execution.
+	 * @return A trigger containing the loaded section. This should be stored and used
+	 * to run the section one or more times.
+	 */
+	@SafeVarargs
+	protected final Trigger loadCode(SectionNode sectionNode, String name, Class<? extends Event>... events) {
+		ParserInstance parser = getParser();
+
+		String previousName = parser.getCurrentEventName();
+		Class<? extends Event>[] previousEvents = parser.getCurrentEvents();
+		SkriptEvent previousSkriptEvent = parser.getCurrentSkriptEvent();
+		List<TriggerSection> previousSections = parser.getCurrentSections();
+		Kleenean previousDelay = parser.getHasDelayBefore();
+
+		parser.setCurrentEvent(name, events);
+		parser.setCurrentSkriptEvent(null);
+		List<TriggerSection> sections = new ArrayList<>();
+		sections.add(this);
+		parser.setCurrentSections(sections);
```

Comment:
I'm not actually sure if we'd want to add this to the current sections, because none of the items loaded will have their `parent` set to this section. 
On the other hand, we do need a way for the items in the section to get this section. 
- Maybe do this with the FakeSkriptEvent, by turning it into a SectionSkriptEvent with a Section field, constructor param and getter? The downside of this is that any syntax element within the section that wants to get this section instance, will have to get it during parsetime instead of runtime, as currentSkriptEvent is only available during parsetime.
- We could also make implementations handle this themselves, for example using their own Bukkit event (the class of which they pass to this method's events parameter) which contains the section instance.

Reply:
Example of an issue with the current implementation:
https://gist.github.com/TPGamesNL/734c75601f74c7c4b2744eb9669e99a1
with the code:
```
command /test:
	trigger:
		do parallel:
			exit
```
prints the error message (`ASSERT FALSE (:`)

Reply:
I think we should try to do what we can to not force this on developers to implement... could we just set the parent ourselves for each item?

Reply:
No, that'd mean the trigger would continue execution after it has ended, and with the wrong event